src/main/java/graphql/schema/diff/SDLSchemaDiffSet.java

```diff
@@ -0,0 +1,74 @@
+package graphql.schema.diff;
+
+import graphql.PublicApi;
+import graphql.language.Document;
+import graphql.parser.Parser;
+import graphql.schema.GraphQLSchema;
+import graphql.schema.idl.SchemaPrinter;
+
+/**
+ * Represents 2 schemas that can be diffed which are defined in SDL.
+ */
+@PublicApi
+public class SDLSchemaDiffSet implements SchemaDiffSet {
+
+    private final String oldSchemaSdl;
+
+    private final String newSchemaSdl;
+
+    public SDLSchemaDiffSet(final String oldSchemaSdl, final String newSchemaSdl) {
+        this.oldSchemaSdl = oldSchemaSdl;
+        this.newSchemaSdl = newSchemaSdl;
+    }
```

Comment:
why store these as Strings and then parse them in the getter.  At this point you cant know they are valid documents

I think it would be better if it was `private SDLSchemaDiffSet(final Document oldSchemaSdl, final Document newSchemaSdl)`

Also make this constructor private.  We have static constructor methods and the change in signature I mentioned above would be possible with private constructors and a breaking change when the they are public.  private constructors on classes should be a starting point

Reply:
Sounds good to me. I do wonder if its worth it to have two classes: SDLSchemaDiffSet and IntrospectionSchemaDiffSet when they both will just be created with documents. I think in this case it would be better to merge it as a single class and just add a note in the builder that introspection will not properly enforce directives?

Reply:
I think this is fine idea